pkg/controller/node/node_controller.go

```diff
@@ -128,6 +143,199 @@ func (ctrl *Controller) Run(workers int, stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+func (crtl *Controller) updateMasterNodes(old, cur interface{}) {
+	oldScheduler := old.(*configv1.Scheduler)
+	curScheduler := cur.(*configv1.Scheduler)
+
+	if oldScheduler.Name != "cluster" && curScheduler.Name != "cluster" {
+		glog.V(4).Infof("We don't care about CRs other than cluster created for scheduler config")
+		return
+	}
+
+	if reflect.DeepEqual(oldScheduler.Spec, curScheduler.Spec) {
+		glog.V(4).Info("Scheduler config did not change")
+		return
+	}
+	desiredSchedulableMasters := curScheduler.Spec.PercentageOfMastersSchedulable
+	currentMasters := crtl.getCurrentMasters()
+	currentMasterCount := len(currentMasters)
+	validateDesiredSchedulableMasterField(desiredSchedulableMasters, currentMasterCount)
+
+	currentSchedulableMasters, currentUnSchedulableMasters := getSchedulableAndUnSchedulabeMasterNodes(currentMasters)
+	desiredNumberOfMastersSchedulable, err := intstrutil.GetValueFromIntOrPercent(intstrutil.ValueOrDefault(&curScheduler.Spec.PercentageOfMastersSchedulable, intstrutil.FromInt(0)), currentMasterCount, false)
+	if err != nil {
+		glog.Errorf("error converting integer")
```

Comment:
This hides the underlying error message; exactly why I prefer `errors.Wrapf`.

Reply:
Sure. Let me look into `errors.Wrapf`. BTW, we can also verify checks in CI which capture this.